// @streamr/cli-tools/dist/bin/streamr-stream-resend-from.js
#!/usr/bin/env node
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
require("../src/logLevel");
const command_1 = require("../src/command");
const resend_1 = require("../src/resend");
(0, command_1.createClientCommand)(async (client, from, streamId, options) => {
    const resendOptions = {
        from: {
            timestamp: Date.parse(from),
            sequenceNumber: 0
        },
        publisherId: options.publisherId
    };
    await (0, resend_1.resend)(streamId, resendOptions, client, options.subscribe);
}, {
    clientOptionsFactory: (options) => ({
        orderMessages: !options.disableOrdering
    })
})
    .arguments('<from> <streamId>')
    .description('request messages starting from given date-time (format: "YYYY-MM-DDTHH:mm:ss.sssZ")')
    .option('--publisher-id <string>', 'filter results by publisher')
    .option('-d, --disable-ordering', 'disable ordering of messages by OrderingUtil', false)
    .option('-s, --subscribe', 'subscribe in addition to resend', false)
    .parseAsync();
